File: app/src/main/java/com/example/m4_individual3/MainActivity.java
package com.example.m4_individual3;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        TextView screen1 = findViewById(R.id.Screen1);
        TextView screen2 = findViewById(R.id.Screen2);
        EditText numUser = findViewById(R.id.numUser);
        Button mas = findViewById(R.id.mas);
        Button resta = findViewById(R.id.resta);
        Button div = findViewById(R.id.div);
        Button por = findViewById(R.id.por);
        Button igual = findViewById(R.id.igual);

        mas.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String txt = numUser.getText().toString();
                screen1.setText(txt);
                screen2.setText("+ ");
                numUser.setText("");
            }
        });

        resta.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String txt = numUser.getText().toString();
                screen1.setText(txt);
                screen2.setText("- ");
                numUser.setText("");
            }
        });

        div.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String txt = numUser.getText().toString();
                screen1.setText(txt);
                screen2.setText("/ ");
                numUser.setText("");
            }
        });

        por.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String txt = numUser.getText().toString();
                screen1.setText(txt);
                screen2.setText("* ");
                numUser.setText("");
            }
        });

        igual.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
            String num1 = screen1.getText().toString();
            String op = screen2.getText().toString().trim();
            String num2 = numUser.getText().toString();
            double resultado =0;
            boolean flagDivCero= true;
            switch (op){
                case "+":
                    resultado= Double.parseDouble(num1) + Double.parseDouble(num2);
                    screen2.setText("+" + num2.toString()) ;
                    break;

                case "-":
                    resultado = Double.parseDouble(num1) - Double.parseDouble(num2);
                    screen2.setText("-" + num2.toString()) ;
                    break;

                case "*":
                    resultado=Double.parseDouble(num1) * Double.parseDouble(num2);
                    screen2.setText("*" + num2.toString()) ;
                    break;

                case "/":
                    if (!num2.equals("0")){
                        resultado= Double.parseDouble(num1) / Double.parseDouble(num2);
                        screen2.setText("/" + num2.toString()) ;
                    }else{
                        flagDivCero=false;
                        Toast.makeText(MainActivity.this, "No se puede dividir por cero", Toast.LENGTH_LONG).show();
                    }
                    break;




            }
                if (flagDivCero){
                    numUser.setText(String.valueOf(resultado));
                }else {
                    numUser.setText("Error al dividir por cero");
                }

            }
        });
    }
}

File: app/src/main/java/com/example/m4_individual3/MainActivity.kt
package com.example.m4_individual3

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val screen1 = findViewById<TextView>(R.id.Screen1)
        val screen2 = findViewById<TextView>(R.id.Screen2)
        val numUser = findViewById<EditText>(R.id.numUser)
        val mas = findViewById<Button>(R.id.mas)
        val resta = findViewById<Button>(R.id.resta)
        val div = findViewById<Button>(R.id.div)
        val por = findViewById<Button>(R.id.por)
        val igual = findViewById<Button>(R.id.igual)
        val exit = findViewById<Button>(R.id.exit)
        val c = findViewById<Button>(R.id.c)
        exit.setOnClickListener { finish() }
        c.setOnClickListener {
            screen1.text = ""
            screen2.text = ""
            numUser.setText("")
        }
        mas.setOnClickListener {
            val txt = numUser.text.toString()
            if (!txt.isEmpty()) {
                screen1.text = txt
                screen2.text = "+ "
                numUser.setText("")
            } else {
                Toast.makeText(this@MainActivity, "Debe Ingresar un Valor", Toast.LENGTH_LONG)
                    .show()
            }
        }
        resta.setOnClickListener {
            val txt = numUser.text.toString()
            if (!txt.isEmpty()) {
                screen1.text = txt
                screen2.text = "- "
                numUser.setText("")
            } else {
                Toast.makeText(this@MainActivity, "Debe Ingresar un Valor", Toast.LENGTH_LONG)
                    .show()
            }
        }
        div.setOnClickListener {
            val txt = numUser.text.toString()
            if (!txt.isEmpty()) {
                screen1.text = txt
                screen2.text = "/ "
                numUser.setText("")
            } else {
                Toast.makeText(this@MainActivity, "Debe Ingresar un Valor", Toast.LENGTH_LONG)
                    .show()
            }
        }
        por.setOnClickListener {
            val txt = numUser.text.toString()
            if (!txt.isEmpty()) {
                screen1.text = txt
                screen2.text = "* "
                numUser.setText("")
            } else {
                Toast.makeText(this@MainActivity, "Debe Ingresar un Valor", Toast.LENGTH_LONG)
                    .show()
            }
        }
        igual.setOnClickListener {
            val num1 = screen1.text.toString()
            val op = screen2.text.toString().trim { it <= ' ' }
            val num2 = numUser.text.toString()
            if (!num2.isEmpty()) {
                var resultado = 0.0
                var flagDivCero = true
                when (op) {
                    "+" -> {
                        resultado = num1.toDouble() + num2.toDouble()
                        screen2.text = "+$num2"
                    }

                    "-" -> {
                        resultado = num1.toDouble() - num2.toDouble()
                        screen2.text = "-$num2"
                    }

                    "*" -> {
                        resultado = num1.toDouble() * num2.toDouble()
                        screen2.text = "*$num2"
                    }

                    "/" -> if (num2 != "0") {
                        resultado = num1.toDouble() / num2.toDouble()
                        screen2.text = "/$num2"
                    } else {
                        flagDivCero = false
                        Toast.makeText(
                            this@MainActivity,
                            "No se puede dividir por cero",
                            Toast.LENGTH_LONG
                        ).show()
                    }
                }
                if (flagDivCero) {
                    numUser.setText(resultado.toString())
                } else {
                    numUser.setText("Error al dividir por cero")
                }
            } else {
                Toast.makeText(this@MainActivity, "Debe Ingresar un Valor", Toast.LENGTH_LONG)
                    .show()
            }
        }
    }
}